ospfd/ospf_sr.c

```diff
@@ -1245,7 +1245,9 @@ static void update_ext_prefix_sid(struct sr_node *srn, struct sr_prefix *srp)
 
 	/* Search for existing Segment Prefix */
 	for (ALL_LIST_ELEMENTS_RO(srn->ext_prefix, node, pref))
-		if (pref->instance == srp->instance) {
+		if (pref->instance == srp->instance
+		    && prefix_same((struct prefix *)&srp->prefv4,
+				   &pref->prefv4)) {
```

Comment:
It is safer, but not necessary. Prefix SID are stored per SR-Node in the SRDB. Here, we search for all prefix of the given SR Node. Second, in OSPF, LSA Opaque instance are unique per router. So, compare instance is sufficient to discriminate prefix assuming that an Extended Prefix Opaque LSA is convey only one prefix SID. If assuming that an Extended Prefix Opaque LSA is convey more than one prefix SID, I agree that comparing prefix is also mandatory for discrimination.